src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/StreamHelpers.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    public static class StreamHelpers
+    {
+        private const int DefaultBufferSize = 1024;
+
+        public static async Task CopyStreamWithNewLineAdded(string[] inputLines, Stream outputStream, string existingLine, string newLine)
```

Comment:
It would be good to have a doc comment here explaining the relationship between existingLine and newLine (newLine is added into the inputLines after any line containing existingLine).